src/assets/scripts/client/aircraft/AircraftModel.js

```diff
@@ -707,8 +707,8 @@ export default class AircraftModel {
             icaoWithWeightClass: this.model.icaoWithWeightClass,
             assignedAltitude,
             flightPlanAltitude,
-            arrivalAirportId: this.destination.toUpperCase(),
-            departureAirportId: this.origin.toUpperCase(),
+            arrivalAirportId: this.destination?.toUpperCase(),
+            departureAirportId: this.origin?.toUpperCase(),
```

Comment:
How do things behave when `origin` is `null`?

Reply:
The `?` operator only calls the following function if the value is not null/undefined/NaN, so the code is equivalent to e.g.:

`this.origin ? this.origin.toUpperCase() : null`

Reply:
Yes, I know. I'm asking because many places in the app use strict `===` equals and this property was formerly a string. An _"invalid"_ or _"empty"_ state before would have been `''`. Now it's null. Using strict equals, what was once true is no longer. 

All I'm asking is: when `origin` or `destination` are [now] `null`, does everything behave the same way? And, if so, can we verify that with a test or two?

Reply:
Sure, I'll take a look over the weekend.

Reply:
I've changed the only place that was comparing the origin and/or destination fields to an empty string in b4fb948, and corrected the defaults to `null`.  I'll take a look at the test coverage later.

Reply:
In terms of test coverage creating objects with default (empty string, now `null`) origin and/or destination is covered both by some of the mocks and some explicit checks for behavior e.g. "_initializeAirportsAndRunways() does not make calls to initialize departure airport+runway when origin ICAO is null".  I updated this in 23dc216 and at first blush I think that should manage expectations within the code appropriately.

Reply:
Awesome! Thanks for looking into that, I appreciate it.

Reply:
I actually caught one more instance of empty strings in the spawn patterns that I missed.  I'll push that tomorrow.